homeassistant/components/fritz/switch.py

```diff
@@ -0,0 +1,633 @@
+"""Switches for AVM Fritz!Box functions."""
+from __future__ import annotations
+
+from collections import OrderedDict
+from functools import partial
+import logging
+from typing import Any, Callable
+
+from fritzconnection.core.exceptions import FritzConnectionException, FritzSecurityError
+import xmltodict
+
+from homeassistant.components.network.util import async_get_source_ip
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import slugify
+
+from .common import FritzBoxTools, SwitchInfo
+from .const import (
+    DOMAIN,
+    SWITCH_TYPE_DEFLECTION,
+    SWITCH_TYPE_DEVICEPROFILE,
+    SWITCH_TYPE_PORTFORWARD,
+    SWITCH_TYPE_WIFINETWORK,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_service_call_action(
```

Comment:
```suggestion
def service_call_action(
```

Then run it directly.

You can make an `async_service_call_action` that is just a wrapper that calls `async_add_executor_job` and `service_call_action`

Reply:
Done ;-)